src/reflect/scala/reflect/internal/util/ScalaClassLoader.scala

```diff
@@ -46,6 +47,35 @@ trait ScalaClassLoader extends JClassLoader {
   def create(path: String): AnyRef =
     tryToInitializeClass[AnyRef](path).map(_.newInstance()).orNull
 
+  /** Create an instance with ctor args, or invoke errorFn before throwing FatalError. */
+  def create[T <: AnyRef : ClassTag](path: String, errorFn: String => Unit)(args: AnyRef*): T = {
+    def fail(msg: String) = {
```

Comment:
OK. Yes, class init errors are classic. You can also see the extra debug at L67, because I needed it.

I'll try to add a, you know, test, as well. I fell asleep before I figured out how.